src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,39 @@
+import { useEffect, useState } from "react";
+import { getArticleById } from "../api";
+import { useParams } from "react-router";
+import SingleArticleCard from "./SingleArticleCard";
+
+export default function SingleArticle() {
+  const [currentArticle, setCurrentArticle] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  const { article_id } = useParams();
+  console.log(article_id, "< id in single article");
+
+  useEffect(() => {
+    setIsLoading(true);
+    getArticleById(article_id)
+      .then(({ article }) => {
+        console.log(article, "< article returned in singlearticlecard");
+        return article;
+      })
+      .then((returnedArticle) => {
+        setCurrentArticle(returnedArticle);
+        setIsError(false);
+        setIsLoading(false);
```

Comment:
this could all be done in the previous then block no need to return and into another.